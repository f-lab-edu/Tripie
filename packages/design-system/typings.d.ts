// declare module '*.module.scss';
declare module '*.module.scss' {
  const classes: { readonly [key: string]: string };
  export default classes;
}

declare module '*.scss' {
  const classes: { readonly [key: string]: string };
  export default classes;
}

declare module '*.css' {
  const content: { [className: string]: string };
  export default content;
}
