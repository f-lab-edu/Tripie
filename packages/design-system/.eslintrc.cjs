/** @type {import("eslint").Linter.Config} */
module.exports = {
  extends: ['@tripie-pyotato/eslint-config/react.js'],
};
