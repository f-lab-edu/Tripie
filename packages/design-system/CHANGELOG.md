# @tripie-pyotato/design-system

## 0.1.0

### Minor Changes

- 394847f: test bundling
