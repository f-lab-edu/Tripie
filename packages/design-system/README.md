# Tripie Design system

> !! THIS LIBRARY IS CURRENTLY UNDERGOING MAJOR CHANGES !!
