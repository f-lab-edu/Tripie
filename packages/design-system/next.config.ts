/**
 * https://nextjs.org/docs/app/building-your-application/styling/sass#customizing-sass-options
 */

// import { fileURLToPath } from 'url';

// const __dirname = fileURLToPath(new URL('.', import.meta.url));
import path from 'path';
/** @type {import('next').NextConfig} */
const nextConfig = {
  sassOptions: {
    includePaths: [path.join(__dirname, 'styles'), path.join(__dirname, '../../packages/design-system/src')],

    silenceDeprecations: ['legacy-js-api'],
  },

  images: {
    dangerouslyAllowSVG: true,
    contentDispositionType: 'attachment',
    contentSecurityPolicy: "default-src 'self'; script-src 'none'; sandbox;",
    remotePatterns: [
      {
        protocol: 'https',
        hostname: 'placehold.co',
        port: '',
        pathname: '/**',
      },
      {
        protocol: 'https',
        hostname: 'eu.ui-avatars.com',
        port: '',
        pathname: '/api/**',
      },
      {
        protocol: 'https',
        hostname: 'media.triple.guide',
        port: '',
        pathname: '/triple-cms/c_limit,f_auto,h_1024,w_1024/**',
      },
      {
        protocol: 'https',
        hostname: 'blogthumb.pstatic.net',
        port: '',
        pathname: '/*',
      },
      {
        protocol: 'http',
        hostname: 'blogthumb.pstatic.net',
        port: '',
        pathname: '/**',
      },

      {
        protocol: 'http',
        hostname: 'blogthumb2.naver.net',
        port: '',
        pathname: '/**',
      },
      {
        protocol: 'https',
        hostname: 'blogthumb2.naver.net',
        port: '',
        pathname: '/**',
      },
      {
        protocol: 'https',
        hostname: 'media.triple.guide',
        port: '',
        pathname: '/triple-cms/**',
      },
      {
        protocol: 'https',
        hostname: 'continentl.com',
        port: '',
        pathname: '/storage/country_flag_image/**',
      },
      {
        protocol: 'https',
        hostname: 'upload.wikimedia.org',
        port: '',
        pathname: '/wikipedia/**',
      },
      {
        protocol: 'https',
        hostname: 'res.cloudinary.com',
        port: '',
        pathname: '/**',
      },
    ],
  },
};

export default nextConfig;
