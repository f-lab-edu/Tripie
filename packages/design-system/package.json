{
  "name": "@tripie-pyotato/design-system",
  "version": "0.3.0",
  "description": "트리피에 사용되는 디자인 시스템",
  "main": "index.js",
  "type": "module",
  "sideEffects": false,
  "scripts": {
    "build": "NODE_OPTIONS='--max-old-space-size=12288' tsup",
    "metafile": "tsup --format cjs,esm",
    "build:pkg": "tsc",
    "lint": "eslint '**/*.{js,jsx,ts,tsx}'",
    "lint:fix": "pnpm run lint --fix",
    "generate:component": "turbo gen react-component",
    "test": "vitest",
    "test:watch": "vitest --watch",
    "clean": "rm -rf ./node_modules && rm -rf ./.turbo && rm -rf ./dist && rm -rf ./.next"
  },
  "devDependencies": {
    "@storybook/react": "^8.2.6",
    "@testing-library/jest-dom": "^6.6.3",
    "@testing-library/react": "^16.0.1",
    "@testing-library/user-event": "^14.4.3",
    "@tripie-pyotato/eslint-config": "^0.0.1",
    "@tripie-pyotato/typescript-config": "^0.0.2",
    "@turbo/gen": "^2.0.11",
    "@types/fs-extra": "^11.0.4",
    "@types/jest": "^29.5.12",
    "@types/node": "^20.11.24",
    "@types/react": "^18.3.3",
    "@vitejs/plugin-react": "^4.3.4",
    "autoprefixer": "^10.4.0",
    "babel-jest": "^29.7.0",
    "bundle-scss": "^1.5.4",
    "css-loader": "^7.1.2",
    "csstype": "^3.1.3",
    "esbuild": "^0.25.2",
    "esbuild-sass-plugin": "^3.3.1",
    "fast-glob": "^3.3.3",
    "fs-extra": "^11.3.0",
    "identity-obj-proxy": "^3.0.0",
    "jest": "^29.7.0",
    "lit-element": "^4.1.1",
    "next": "^15.0.3",
    "postcss": "^8.5.3",
    "postcss-cli": "^11.0.1",
    "postcss-copy-assets": "^0.3.1",
    "postcss-modules": "^6.0.1",
    "postcss-utilities": "^0.8.4",
    "react": "^18.3.1",
    "rollup-preserve-directives": "^1.1.3",
    "sass": "^1.85.0",
    "ts-jest": "^29.2.4",
    "tsup": "^8.4.0",
    "tsx": "^4.20.3",
    "vitest": "^3.0.6"
  },
  "dependencies": {
    "@tripie-pyotato/hooks": "^0.0.3",
    "@tripie-pyotato/utils": "^0.0.3",
    "classnames": "^2.5.1",
    "date-fns": "^4.1.0"
  },
  "typesVersions": {
    "*": {
      "*": [
        "src/*",
        "src/*/index"
      ]
    }
  },
  "peerDependencies": {
    "@base-ui-components/react": "1.0.0-beta.0",
    "framer": "^2.4.1",
    "framer-motion": "^11.11.17",
    "maplibre-gl": "^4.7.1",
    "next": "*",
    "react": "*",
    "react-calendar": "^5.0.0",
    "react-dom": "^18.3.1",
    "react-globe.gl": "^2.27.2",
    "react-intersection-observer": "9.13.1",
    "react-map-gl": "^7.1.7",
    "usehooks-ts": "^3.1.0"
  },
  "exports": {
    ".": {
      "require": {
        "types": "./dist/index.d.cts",
        "default": "./dist/index.cjs"
      },
      "import": {
        "types": "./dist/index.d.ts",
        "default": "./dist/index.js"
      }
    },
    "./shared": {
      "require": {
        "types": "./dist/shared/index.d.cts",
        "default": "./dist/shared/index.cjs"
      },
      "import": {
        "types": "./dist/shared/index.d.ts",
        "default": "./dist/shared/index.js"
      }
    },
    "./provider": {
      "require": {
        "types": "./dist/provider/index.d.cts",
        "default": "./dist/provider/index.cjs"
      },
      "import": {
        "types": "./dist/provider/index.d.ts",
        "default": "./dist/provider/index.js"
      }
    },
    "./global": {
      "require": {
        "default": "./dist/global.css"
      },
      "import": {
        "default": "./dist/global.css"
      }
    },
    "./font": {
      "require": {
        "default": "./dist/font.css"
      },
      "import": {
        "default": "./dist/font.css"
      }
    },
    "./styles": {
      "import": "./dist/index.css",
      "require": "./dist/index.css",
      "default": "./dist/index.css"
    },
    "./@hooks": {
      "require": {
        "types": "./dist/@hooks/index.d.cts",
        "default": "./dist/@hooks/index.cjs"
      },
      "import": {
        "types": "./dist/@hooks/index.d.ts",
        "default": "./dist/@hooks/index.js"
      }
    },
    "./@core": {
      "require": {
        "types": "./dist/@core/index.d.cts",
        "default": "./dist/@core/index.cjs"
      },
      "import": {
        "types": "./dist/@core/index.d.ts",
        "default": "./dist/@core/index.js"
      }
    },
    "./@core/data-display": {
      "require": {
        "types": "./dist/@core/data-display/index.d.cts",
        "default": "./dist/@core/data-display/index.cjs"
      },
      "import": {
        "types": "./dist/@core/data-display/index.d.ts",
        "default": "./dist/@core/data-display/index.js"
      }
    },
    "./@core/feedback": {
      "require": {
        "types": "./dist/@core/feedback/index.d.cts",
        "default": "./dist/@core/feedback/index.cjs"
      },
      "import": {
        "types": "./dist/@core/feedback/index.d.ts",
        "default": "./dist/@core/feedback/index.js"
      }
    },
    "./@core/layout": {
      "require": {
        "types": "./dist/@core/layout/index.d.cts",
        "default": "./dist/@core/layout/index.cjs"
      },
      "import": {
        "types": "./dist/@core/layout/index.d.ts",
        "default": "./dist/@core/layout/index.js"
      }
    },
    "./@components": {
      "require": {
        "types": "./dist/@components/index.d.cts",
        "default": "./dist/@components/index.cjs"
      },
      "import": {
        "types": "./dist/@components/index.d.ts",
        "default": "./dist/@components/index.js"
      }
    },
    "./@components/data-display": {
      "require": {
        "types": "./dist/@components/data-display/index.d.cts",
        "default": "./dist/@components/data-display/index.cjs"
      },
      "import": {
        "types": "./dist/@components/data-display/index.d.ts",
        "default": "./dist/@components/data-display/index.js"
      }
    },
    "./@components/feedback": {
      "require": {
        "types": "./dist/@components/feedback/index.d.cts",
        "default": "./dist/@components/feedback/index.cjs"
      },
      "import": {
        "types": "./dist/@components/feedback/index.d.ts",
        "default": "./dist/@components/feedback/index.js"
      }
    },
    "./@components/inputs": {
      "require": {
        "types": "./dist/@components/inputs/index.d.cts",
        "default": "./dist/@components/inputs/index.cjs"
      },
      "import": {
        "types": "./dist/@components/inputs/index.d.ts",
        "default": "./dist/@components/inputs/index.js"
      }
    },
    "./@components/navigation": {
      "require": {
        "types": "./dist/@components/navigation/index.d.cts",
        "default": "./dist/@components/navigation/index.cjs"
      },
      "import": {
        "types": "./dist/@components/navigation/index.d.ts",
        "default": "./dist/@components/navigation/index.js"
      }
    },
    "./@components/surfaces": {
      "require": {
        "types": "./dist/@components/surfaces/index.d.cts",
        "default": "./dist/@components/surfaces/index.cjs"
      },
      "import": {
        "types": "./dist/@components/surfaces/index.d.ts",
        "default": "./dist/@components/surfaces/index.js"
      }
    },
    "./@components/x": {
      "require": {
        "types": "./dist/@components/x/index.d.cts",
        "default": "./dist/@components/x/index.cjs"
      },
      "import": {
        "types": "./dist/@components/x/index.d.ts",
        "default": "./dist/@components/x/index.js"
      }
    },
    "./@wrappers": {
      "require": {
        "types": "./dist/wrappers/index.d.cts",
        "default": "./dist/wrappers/index.cjs"
      },
      "import": {
        "types": "./dist/wrappers/index.d.ts",
        "default": "./dist/wrappers/index.js"
      }
    },
    "./@components/AnimatedText": {
      "require": {
        "types": "./dist/@components/data-display/AnimatedText/index.d.cts",
        "default": "./dist/@components/data-display/AnimatedText/AnimatedText.client.cjs"
      },
      "import": {
        "types": "./dist/@components/data-display/AnimatedText/index.d.ts",
        "default": "./dist/@components/data-display/AnimatedText/AnimatedText.client.js"
      }
    },
    "./@components/Carousel": {
      "require": {
        "types": "./dist/@components/data-display/Carousel/index.d.cts",
        "default": "./dist/@components/data-display/Carousel/Carousel.client.cjs"
      },
      "import": {
        "types": "./dist/@components/data-display/Carousel/index.d.ts",
        "default": "./dist/@components/data-display/Carousel/Carousel.client.js"
      }
    },
    "./@components/Chip": {
      "require": {
        "types": "./dist/@components/data-display/Chip/index.d.cts",
        "default": "./dist/@components/data-display/Chip/Chip.client.cjs"
      },
      "import": {
        "types": "./dist/@components/data-display/Chip/index.d.ts",
        "default": "./dist/@components/data-display/Chip/Chip.client.js"
      }
    },
    "./@components/Divider": {
      "require": {
        "types": "./dist/@components/data-display/Divider/index.d.cts",
        "default": "./dist/@components/data-display/Divider/Divider.client.cjs"
      },
      "import": {
        "types": "./dist/@components/data-display/Divider/index.d.ts",
        "default": "./dist/@components/data-display/Divider/Divider.client.js"
      }
    },
    "./@components/Image": {
      "require": {
        "types": "./dist/@components/data-display/Image/index.d.cts",
        "default": "./dist/@components/data-display/Image/Image.client.cjs"
      },
      "import": {
        "types": "./dist/@components/data-display/Image/index.d.ts",
        "default": "./dist/@components/data-display/Image/Image.client.js"
      }
    },
    "./@components/TextFillAnimation": {
      "require": {
        "types": "./dist/@components/data-display/TextFillAnimation/index.d.cts",
        "default": "./dist/@components/data-display/TextFillAnimation/TextFillAnimation.client.cjs"
      },
      "import": {
        "types": "./dist/@components/data-display/TextFillAnimation/index.d.ts",
        "default": "./dist/@components/data-display/TextFillAnimation/TextFillAnimation.client.js"
      }
    },
    "./@components/TextUnderlineAnimation": {
      "require": {
        "types": "./dist/@components/data-display/TextUnderlineAnimation/index.d.cts",
        "default": "./dist/@components/data-display/TextUnderlineAnimation/TextUnderlineAnimation.client.cjs"
      },
      "import": {
        "types": "./dist/@components/data-display/TextUnderlineAnimation/index.d.ts",
        "default": "./dist/@components/data-display/TextUnderlineAnimation/TextUnderlineAnimation.client.js"
      }
    },
    "./@components/Tooltip": {
      "require": {
        "types": "./dist/@components/data-display/Tooltip/index.d.cts",
        "default": "./dist/@components/data-display/Tooltip/Tooltip.client.cjs"
      },
      "import": {
        "types": "./dist/@components/data-display/Tooltip/index.d.ts",
        "default": "./dist/@components/data-display/Tooltip/Tooltip.client.js"
      }
    },
    "./@components/Icon": {
      "require": {
        "types": "./dist/@components/data-display/TripieIcon/index.d.cts",
        "default": "./dist/@components/data-display/TripieIcon/index.cjs"
      },
      "import": {
        "types": "./dist/@components/data-display/TripieIcon/index.d.ts",
        "default": "./dist/@components/data-display/TripieIcon/index.js"
      }
    },
    "./@components/Notification": {
      "require": {
        "types": "./dist/@components/feedback/Notification/index.d.cts",
        "default": "./dist/@components/feedback/Notification/Notification.client.cjs"
      },
      "import": {
        "types": "./dist/@components/feedback/Notification/index.d.ts",
        "default": "./dist/@components/feedback/Notification/Notification.client.js"
      }
    },
    "./@components/SplashScreen": {
      "require": {
        "types": "./dist/@components/feedback/SplashScreen/index.d.cts",
        "default": "./dist/@components/feedback/SplashScreen/SplashScreen.client.cjs"
      },
      "import": {
        "types": "./dist/@components/feedback/SplashScreen/index.d.ts",
        "default": "./dist/@components/feedback/SplashScreen/SplashScreen.client.js"
      }
    },
    "./@components/Toast": {
      "require": {
        "types": "./dist/@components/feedback/Toast/index.d.cts",
        "default": "./dist/@components/feedback/Toast/Toast.cjs"
      },
      "import": {
        "types": "./dist/@components/feedback/Toast/index.d.ts",
        "default": "./dist/@components/feedback/Toast/Toast.js"
      }
    },
    "./@components/Switch": {
      "require": {
        "types": "./dist/@components/inputs/Switch/index.d.cts",
        "default": "./dist/@components/inputs/Switch/Switch.client.cjs"
      },
      "import": {
        "types": "./dist/@components/inputs/Switch/index.d.ts",
        "default": "./dist/@components/inputs/Switch/Switch.client.js"
      }
    },
    "./@components/Button": {
      "require": {
        "types": "./dist/@components/inputs/TripieButton/index.d.cts",
        "default": "./dist/@components/inputs/TripieButton/index.cjs"
      },
      "import": {
        "types": "./dist/@components/inputs/TripieButton/index.d.ts",
        "default": "./dist/@components/inputs/TripieButton/index.js"
      }
    },
    "./@components/FlickTextButton": {
      "require": {
        "types": "./dist/@components/inputs/TripieButton/Flick/index.d.cts",
        "default": "./dist/@components/inputs/TripieButton/Flick/Flick.client.cjs"
      },
      "import": {
        "types": "./dist/@components/inputs/TripieButton/Flick/index.d.ts",
        "default": "./dist/@components/inputs/TripieButton/Flick/Flick.client.js"
      }
    },
    "./@components/AccentedButton": {
      "require": {
        "types": "./dist/@components/inputs/TripieButton/Accented/index.d.cts",
        "default": "./dist/@components/inputs/TripieButton/Accented/Accented.client.cjs"
      },
      "import": {
        "types": "./dist/@components/inputs/TripieButton/Accented/index.d.ts",
        "default": "./dist/@components/inputs/TripieButton/Accented/Accented.client.js"
      }
    },
    "./@components/AnimatedButton": {
      "require": {
        "types": "./dist/@components/inputs/TripieButton/Animated/index.d.cts",
        "default": "./dist/@components/inputs/TripieButton/Animated/AnimatedButton.client.cjs"
      },
      "import": {
        "types": "./dist/@components/inputs/TripieButton/Animated/index.d.ts",
        "default": "./dist/@components/inputs/TripieButton/Animated/AnimatedButton.client.js"
      }
    },
    "./@components/BasicButton": {
      "require": {
        "types": "./dist/@components/inputs/TripieButton/Basic/index.d.cts",
        "default": "./dist/@components/inputs/TripieButton/Basic/BasicButton.client.cjs"
      },
      "import": {
        "types": "./dist/@components/inputs/TripieButton/Basic/index.d.ts",
        "default": "./dist/@components/inputs/TripieButton/Basic/BasicButton.client.js"
      }
    },
    "./@components/NavigationButton": {
      "require": {
        "types": "./dist/@components/navigation/Button/index.d.cts",
        "default": "./dist/@components/navigation/Button/NavigationButton.client.cjs"
      },
      "import": {
        "types": "./dist/@components/navigation/Button/index.d.ts",
        "default": "./dist/@components/navigation/Button/NavigationButton.client.js"
      }
    },
    "./@components/Drawer": {
      "require": {
        "types": "./dist/@components/navigation/Drawer/index.d.cts",
        "default": "./dist/@components/navigation/Drawer/Drawer.client.cjs"
      },
      "import": {
        "types": "./dist/@components/navigation/Drawer/index.d.ts",
        "default": "./dist/@components/navigation/Drawer/Drawer.client.js"
      }
    },
    "./@components/Link": {
      "require": {
        "types": "./dist/@components/navigation/Link/index.d.cts",
        "default": "./dist/@components/navigation/Link/Link.client.cjs"
      },
      "import": {
        "types": "./dist/@components/navigation/Link/index.d.ts",
        "default": "./dist/@components/navigation/Link/Link.client.js"
      }
    },
    "./@components/Menu": {
      "require": {
        "types": "./dist/@components/navigation/Menu/index.d.cts",
        "default": "./dist/@components/navigation/Menu/Menu.client.cjs"
      },
      "import": {
        "types": "./dist/@components/navigation/Menu/index.d.ts",
        "default": "./dist/@components/navigation/Menu/Menu.client.js"
      }
    },
    "./@components/Accordion": {
      "require": {
        "types": "./dist/@components/surfaces/Accordion/index.d.cts",
        "default": "./dist/@components/surfaces/Accordion/Accordion.client.cjs"
      },
      "import": {
        "types": "./dist/@components/surfaces/Accordion/index.d.ts",
        "default": "./dist/@components/surfaces/Accordion/Accordion.client.js"
      }
    },
    "./@components/AnimatedCard": {
      "require": {
        "types": "./dist/@components/surfaces/AnimatedCard/index.d.cts",
        "default": "./dist/@components/surfaces/AnimatedCard/AnimatedCard.client.cjs"
      },
      "import": {
        "types": "./dist/@components/surfaces/AnimatedCard/index.d.ts",
        "default": "./dist/@components/surfaces/AnimatedCard/AnimatedCard.client.js"
      }
    },
    "./@components/AnimatedContainer": {
      "require": {
        "types": "./dist/@components/surfaces/AnimatedContainer/index.d.cts",
        "default": "./dist/@components/surfaces/AnimatedContainer/AnimatedContainer.client.cjs"
      },
      "import": {
        "types": "./dist/@components/surfaces/AnimatedContainer/index.d.ts",
        "default": "./dist/@components/surfaces/AnimatedContainer/AnimatedContainer.client.js"
      }
    },
    "./@components/AppBar": {
      "require": {
        "types": "./dist/@components/surfaces/AppBar/index.d.cts",
        "default": "./dist/@components/surfaces/AppBar/AppBar.client.cjs"
      },
      "import": {
        "types": "./dist/@components/surfaces/AppBar/index.d.ts",
        "default": "./dist/@components/surfaces/AppBar/AppBar.client.js"
      }
    },
    "./@components/Card": {
      "require": {
        "types": "./dist/@components/surfaces/Card/index.d.cts",
        "default": "./dist/@components/surfaces/Card/Card.client.cjs"
      },
      "import": {
        "types": "./dist/@components/surfaces/Card/index.d.ts",
        "default": "./dist/@components/surfaces/Card/Card.client.js"
      }
    },
    "./@components/Calendar": {
      "require": {
        "types": "./dist/@components/x/Calendar/index.d.cts",
        "default": "./dist/@components/x/Calendar/Calendar.client.cjs"
      },
      "import": {
        "types": "./dist/@components/x/Calendar/index.d.ts",
        "default": "./dist/@components/x/Calendar/Calendar.client.js"
      }
    },
    "./@components/Globe": {
      "require": {
        "types": "./dist/@components/x/Globe/index.d.cts",
        "default": "./dist/@components/x/Globe/RotatingGlobe.client.cjs"
      },
      "import": {
        "types": "./dist/@components/x/Globe/index.d.ts",
        "default": "./dist/@components/x/Globe/RotatingGlobe.client.js"
      }
    },
    "./@components/MapLines": {
      "require": {
        "types": "./dist/@components/x/Lines/index.d.cts",
        "default": "./dist/@components/x/Lines/Lines.client.cjs"
      },
      "import": {
        "types": "./dist/@components/x/Lines/index.d.ts",
        "default": "./dist/@components/x/Lines/Lines.client.js"
      }
    },
    "./@components/Map": {
      "require": {
        "types": "./dist/@components/x/Map/index.d.cts",
        "default": "./dist/@components/x/Map/Map.client.cjs"
      },
      "import": {
        "types": "./dist/@components/x/Map/index.d.ts",
        "default": "./dist/@components/x/Map/Map.client.js"
      }
    },
    "./@components/MapMarker": {
      "require": {
        "types": "./dist/@components/x/Marker/index.d.cts",
        "default": "./dist/@components/x/Marker/index.cjs"
      },
      "import": {
        "types": "./dist/@components/x/Marker/index.d.ts",
        "default": "./dist/@components/x/Marker/index.js"
      }
    },
    "./@components/MapPopup": {
      "require": {
        "types": "./dist/@components/x/MapPopup/index.d.cts",
        "default": "./dist/@components/x/MapPopup/Popup.client.cjs"
      },
      "import": {
        "types": "./dist/@components/x/MapPopup/index.d.ts",
        "default": "./dist/@components/x/MapPopup/Popup.client.js"
      }
    },
    "./@components/MotionSlideUp": {
      "require": {
        "types": "./dist/@components/x/MotionSlideUp/index.d.cts",
        "default": "./dist/@components/x/MotionSlideUp/MotionSlideUp.client.cjs"
      },
      "import": {
        "types": "./dist/@components/x/MotionSlideUp/index.d.ts",
        "default": "./dist/@components/x/MotionSlideUp/MotionSlideUp.client.js"
      }
    },
    "./@components/ParticleBackground": {
      "require": {
        "types": "./dist/@components/x/Particle/Background/index.d.cts",
        "default": "./dist/@components/x/Particle/Background/ParticleBackground.client.cjs"
      },
      "import": {
        "types": "./dist/@components/x/Particle/Background/index.d.ts",
        "default": "./dist/@components/x/Particle/Background/ParticleBackground.client.js"
      }
    },
    "./@components/ParticleField": {
      "require": {
        "types": "./dist/@components/x/Particle/Field/index.d.cts",
        "default": "./dist/@components/x/Particle/Field/ParticleField.client.cjs"
      },
      "import": {
        "types": "./dist/@components/x/Particle/Field/index.d.ts",
        "default": "./dist/@components/x/Particle/Field/ParticleField.client.js"
      }
    },
    "./@components/RotatingBlur": {
      "require": {
        "types": "./dist/@components/x/RotatingBlur/index.d.cts",
        "default": "./dist/@components/x/RotatingBlur/RotatingBlur.client.cjs"
      },
      "import": {
        "types": "./dist/@components/x/RotatingBlur/index.d.ts",
        "default": "./dist/@components/x/RotatingBlur/RotatingBlur.client.js"
      }
    },
    "./@core/Headings": {
      "require": {
        "types": "./dist/@cor/data-display/Headings/index.d.cts",
        "default": "./dist/@core/data-display/Headings/Headings.cjs"
      },
      "import": {
        "types": "./dist/@core/data-display/Headings/index.d.ts",
        "default": "./dist/@core/data-display/Headings/Headings.js"
      }
    },
    "./@core/List": {
      "require": {
        "types": "./dist/@cor/data-display/List/index.d.cts",
        "default": "./dist/@core/data-display/List/List.cjs"
      },
      "import": {
        "types": "./dist/@core/data-display/List/index.d.ts",
        "default": "./dist/@core/data-display/List/List.js"
      }
    },
    "./@core/ListItem": {
      "require": {
        "types": "./dist/@cor/data-display/ListItem/index.d.cts",
        "default": "./dist/@core/data-display/ListItem/ListItem.cjs"
      },
      "import": {
        "types": "./dist/@core/data-display/ListItem/index.d.ts",
        "default": "./dist/@core/data-display/ListItem/ListItem.js"
      }
    },
    "./@core/Text": {
      "require": {
        "types": "./dist/@cor/data-display/Text/index.d.cts",
        "default": "./dist/@core/data-display/Text/index.cjs"
      },
      "import": {
        "types": "./dist/@core/data-display/Text/index.d.ts",
        "default": "./dist/@core/data-display/Text/index.js"
      }
    },
    "./@core/Skeleton": {
      "require": {
        "types": "./dist/@cor/feedback/TripieSkeleton/index.d.cts",
        "default": "./dist/@core/feedback/TripieSkeleton/Skeleton.cjs"
      },
      "import": {
        "types": "./dist/@core/feedback/TripieSkeleton/index.d.ts",
        "default": "./dist/@core/feedback/TripieSkeleton/Skeleton.js"
      }
    },
    "./@core/Background": {
      "require": {
        "types": "./dist/@cor/layout/Background/index.d.cts",
        "default": "./dist/@core/layout/Background/TripieBackground.cjs"
      },
      "import": {
        "types": "./dist/@core/layout/Background/index.d.ts",
        "default": "./dist/@core/layout/Background/TripieBackground.js"
      }
    },
    "./@core/Stack": {
      "require": {
        "types": "./dist/@cor/layout/Stack/index.d.cts",
        "default": "./dist/@core/layout/Stack/Stack.cjs"
      },
      "import": {
        "types": "./dist/@core/layout/Stack/index.d.ts",
        "default": "./dist/@core/layout/Stack/Stack.js"
      }
    },
    "./@core/Table": {
      "require": {
        "types": "./dist/@cor/layout/Table/index.d.cts",
        "default": "./dist/@core/layout/Table/Table.cjs"
      },
      "import": {
        "types": "./dist/@core/layout/Table/index.d.ts",
        "default": "./dist/@core/layout/Table/Table.js"
      }
    },
    "./@core/Container": {
      "require": {
        "types": "./dist/@core/layout/TripieContainer/index.d.cts",
        "default": "./dist/@core/layout/TripieContainer/TripieContainer.cjs"
      },
      "import": {
        "types": "./dist/@core/layout/TripieContainer/index.d.ts",
        "default": "./dist/@core/layout/TripieContainer/TripieContainer.js"
      }
    }
  }
}
