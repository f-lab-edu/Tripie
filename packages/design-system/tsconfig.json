{
  "compilerOptions": {
    "rootDir": ".",
    "outDir": "./dist",
    "jsx": "preserve",
    "baseUrl": "./src",
    "paths": {
      "@components/*": ["components/*"]
    }
  },
  "include": [
    "**/*.ts",
    "**/*.tsx",
    "../../typings",
    "jest.setup.cjs",
    "~/*.ts",
    "~/*.tsx",
    "src/shared/wrappers/index.tsz"
  ],
  "exclude": ["dist"]
}
