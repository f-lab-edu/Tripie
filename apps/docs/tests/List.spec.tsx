import { render, screen } from '@testing-library/react';
import { Link } from '@tripie-pyotato/design-system';
import { beforeAll, describe, expect, test } from 'vitest';

describe('UnstyledLink Component', () => {
  beforeAll(() => {
    // âœ… Ensure `document` is available before running tests
    global.document = window.document;
  });

  test('renders the link with the correct text and href', () => {
    render(<Link href="/home">Go Home</Link>);

    const linkElement = screen.getByRole('link', { name: /go home/i });

    expect(linkElement).toBeDefined();
    expect(linkElement).toMatchSnapshot();
  });
});
