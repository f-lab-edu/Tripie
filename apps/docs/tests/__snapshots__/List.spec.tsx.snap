// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UnstyledLink Component > renders the link with the correct text and href 1`] = `
<a
  class="_link_1izjp_382"
  href="/home"
>
  Go Home
</a>
`;
